linera-execution/src/evm/revm.rs

```diff
@@ -706,11 +732,19 @@ impl<Runtime: ContractRuntime> CallInterceptorContract<Runtime> {
         context: &mut Ctx<'_, Runtime>,
         inputs: &mut CallInputs,
     ) -> Result<Option<CallOutcome>, ExecutionError> {
-        let contract_address = Address::ZERO.create(0);
-        if inputs.target_address == PRECOMPILE_ADDRESS || inputs.target_address == contract_address
+        // Every call to a contract passes by this function.
+        // Three kinds:
+        // --- Call to the PRECOMPILE smart contract.
+        // --- Call to the EVM smart contract itself
+        // --- Call to other EVM smart contract
+        if self.precompile_addresses.contains(&inputs.target_address)
+            || inputs.target_address == self.contract_address
         {
+            // Precompile calls are handled by the precompile code.
+            // The EVM smart contract is being called
             return Ok(None);
         }
+        // Other smart contracts calls are handled by the runtime
         let target = address_to_user_application_id(inputs.target_address);
         let argument = get_call_argument(context, &inputs.input);
         let authenticated = true;
```

Comment:
Not a change from this PR but:

why is `authenticated` set to `true` here? In Wasm contracts it's an argument of a call and not hardcoded.

Reply:
It is a difference between Linera and the EVM.